reclass/datatypes/parameters.py

```diff
@@ -134,19 +153,42 @@ def _merge_dict(self, cur, new, path):
             ret.update(new)
             return ret
 
+        ovrprfx = Parameters.DICT_KEY_OVERRIDE_PREFIX
+
         for key, newvalue in new.iteritems():
-            ret[key] = self._merge_recurse(ret.get(key), newvalue,
-                                           path.new_subpath(key))
+            if key.startswith(ovrprfx) and not initmerge:
+                ret[key.lstrip(ovrprfx)] = newvalue
+            else:
```

Comment:
I think that the case `if key.startswith(overprfx) and initmerge` should be explicitly handled. I don't think we want to allow processed parameter keys to contain special characters without those characters being explicitly escaped.